app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb

```diff
@@ -0,0 +1,160 @@
+module Authentication
+  module AuthnJwt
+    module VendorConfigurations
+      # Mock JWTConfiguration class to use it to develop other part in the jwt authenticator
+      class ConfigurationJWTGenericVendor < ConfigurationInterface
+
+        def initialize(authenticator_input)
+          super
+
+          @logger = Rails.logger
+
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CreatingAuthenticationParametersObject.new)
+          @authentication_parameters = authentication_parameters_class.new(
+            authentication_input: authenticator_input,
+            jwt_token: jwt_token(authenticator_input)
+          )
+        end
+
+        def jwt_identity
+          @jwt_identity ||= jwt_identity_from_request
+        end
+
+        def validate_restrictions
+          validate_resource_restrictions.call(
+            authenticator_name: @authentication_parameters.authenticator_name,
+            service_id: @authentication_parameters.service_id,
+            account: @authentication_parameters.account,
+            role_name: jwt_identity,
+            constraints: constraints,
+            authentication_request: restriction_validator_class.new(
+              decoded_token: @authentication_parameters.decoded_token
+            )
+          )
+        end
+
+        def validate_and_decode_token
+          @authentication_parameters.decoded_token = validate_and_decode_token_instance.call(
+            authentication_parameters: @authentication_parameters
+          )
+        end
+
+        private
+
+        def jwt_token(authenticator_input)
+          extract_token_from_credentials.call(
+            credentials: authenticator_input.request.body.read
+          )
+        end
+
+        def jwt_identity_from_request
+          @jwt_identity_from_request = identity_provider.jwt_identity
+        end
+
+        def identity_provider
+          @identity_provider = create_identity_provider.call(
+            authentication_parameters: @authentication_parameters
+          )
+        end
+
+        def validate_and_decode_token_instance
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CreateValidateAndDecodeTokenInstance.new)
+          @validate_and_decode_token_instance ||= validate_and_decode_token_class.new(
+            fetch_signing_key: fetch_signing_key,
+            fetch_jwt_claims_to_validate: fetch_jwt_claims_to_validate
+          )
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CreatedValidateAndDecodeTokenInstance.new)
+          @validate_and_decode_token_instance
+        end
```

Comment:
currently you set `@validate_and_decode_token_instance` once but you log the message every time you call the method. please change it to:
```suggestion
        def validate_and_decode_token_instance
          return @validate_and_decode_token_instance if @validate_and_decode_token_instance
          
          @logger.debug(LogMessages::Authentication::AuthnJwt::CreateValidateAndDecodeTokenInstance.new)
          @validate_and_decode_token_instance = validate_and_decode_token_class.new(
            fetch_signing_key: fetch_signing_key,
            fetch_jwt_claims_to_validate: fetch_jwt_claims_to_validate
          )
          @logger.debug(LogMessages::Authentication::AuthnJwt::CreatedValidateAndDecodeTokenInstance.new)
          @validate_and_decode_token_instance
        end
```